[Vanessa Mae] iP

Vanessamae23 opened the pull request:
# Duke Project Pro
> “Your mind is for having ideas, not holding them.” – David Allen

DukePro frees your mind of having to remember things you need to do. It's,
- Text-based
- Best way to learn
- Easy and Simple to use

All you need to do is,

1. Download from [link](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. Double click it
3. Add your tasks
4. Let it manage your task for you :blush:
It is also *Emphasize* Free

Features
- [ ]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```



## Setting up in Intellij

Prerequisites: JDK 11, update Intellij to the most recent version.

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)
1. Open the project into Intellij as follows:
   1. Click `Open`.
   1. Select the project directory, and click `OK`.
   1. If there are any further prompts, accept the defaults.
1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
   In the same dialog, set the **Project language level** field to the `SDK default` option.
3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
   ```
   Hello from
    ____        _        
   |  _ \ _   _| | _____ 
   | | | | | | | |/ / _ \
   | |_| | |_| |   <  __/
   |____/ \__,_|_|\_\___|
   ```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Vanessamae23 committed:
Added level 0 tags

Vanessamae23 committed:
Added level 1 tag

Vanessamae23 committed:
Added feature for level 1 tag

Vanessamae23 committed:
Added level 2

Vanessamae23 committed:
Added level 3

Vanessamae23 committed:
Added level 4 and fixed sth in level 0

Vanessamae23 committed:
Added testing

Vanessamae23 committed:
Added exception handling

Vanessamae23 committed:
Added level 6

Vanessamae23 committed:
Added enums

Vanessamae23 committed:
Add level 7 requirements

Vanessamae23 committed:
Fix merging issues

Vanessamae23 committed:
Add level 8

Vanessamae23 committed:
resolve conflicts

Vanessamae23 committed:
More OOP

Vanessamae23 committed:
Add package

Vanessamae23 committed:
fix issue

Vanessamae23 committed:
Merge branch 'add-gradle-support'

Vanessamae23 committed:
Add testing

Vanessamae23 committed:
Added javadoc

Vanessamae23 committed:
Add code standards

Vanessamae23 committed:
Add find method

Vanessamae23 committed:
After merging two branch

Vanessamae23 committed:
After merge last

Vanessamae23 committed:
After merge last

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Added checkstyle

Vanessamae23 committed:
Merge branch 'master' of https://github.com/Vanessamae23/cs2103t-ip

Vanessamae23 committed:
Added javafx

ngeeyonglim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,199 @@
+import javafx.scene.layout.Region;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
```
Good use of imports and grouping them.

ngeeyonglim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,199 @@
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
```
Good variable names, follows standards.

ngeeyonglim commented on src/main/java/seedu/Command.java:
```diff
@@ -0,0 +1,61 @@
+public class Command {
+    private String command;
+    private Ui ui;
+    private boolean exit = false;
```
Perhaps you can use isExit instead of exit?

ngeeyonglim commented on src/main/java/seedu/Command.java:
```diff
@@ -0,0 +1,61 @@
+                this.ui.showMarked();
+            } else if (command.startsWith("find")) {
+                String keyword = command.substring(4).trim();
+                ArrayList<Task> foundWords = this.tasks.find(keyword);
```
Good variable name for array

ngeeyonglim commented on src/main/java/seedu/Storage.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.List;
+
+public class Storage {
+    private final Path filePath;
```
Perhaps you would like to change this variable to FILE_PATH

ngeeyonglim commented on src/main/java/seedu/Task.java:
```diff
@@ -0,0 +1,121 @@
+        this.isDone = false;
+        if (category.equals("todo")) {
+            this.category = Type.ToDo;
+            String[] splitWord =description.split(" ", 2);
```
You might want to check the spacing here

ngeeyonglim reviewed (commented):
Overall very well done and clear and consider code. Can check some of the variable names and spacing in the code to make sure it is up to the coding standard.

Vanessamae23 committed:
Added a new branch

Vanessamae23 committed:
Made changes

nknguyenhc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,194 @@
-public class Duke {
+package seedu;
```
`Duke` class is not in `seedu` folder and hence should not be in `seedu` package. This can cause troubles when using Gradle to run the programme.

nknguyenhc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,194 @@
+import javafx.scene.layout.Region;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
```
Imports should follow lexical graphical order. Generally, you can run `gradlew checkstyleMain` to make sure that your code follows the course's coding standard.

nknguyenhc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,194 @@
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
```
Once the method is deleted, the Javadoc comment should be deleted as well.

nknguyenhc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,194 @@
+            ArrayList<String> s = new ArrayList<>();
+            s = c.execute();
+            return s.toArray(new String[0]);
+        } catch (Exception e) {
```
This can potentially lead to bug masking. It is better to catch only specific types of exceptions?

nknguyenhc commented on src/main/java/seedu/Parser.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class Parser {
+    public static Command parse(String command, Ui ui, Storage storage, TaskList tasks) {
+        
```
Empty lines after an opening bracket can be cut.

nknguyenhc commented on src/main/java/seedu/Parser.java:
```diff
@@ -0,0 +1,28 @@
+public class Parser {
+    public static Command parse(String command, Ui ui, Storage storage, TaskList tasks) {
+        
+        if(command.equals("bye")) {
```
There should be white space after the `if` keyword.

nknguyenhc commented on src/main/java/seedu/Storage.java:
```diff
@@ -0,0 +1,38 @@
+        try {
+            List<String> tasks = Files.readAllLines(filePath);
+            return TaskList.getFromFile(tasks);
+        } catch(IOException e) {
```
There should be a white space after the `catch` keyword.

nknguyenhc commented on src/main/java/seedu/Task.java:
```diff
@@ -0,0 +1,121 @@
+                    this.description = title + "("+ splitWord[1].split("by ")[1].trim() + ")";
+                } else {
+                    String date = splitWord[1].substring( 3).length() == 1 ? "0"
+                            + splitWord[1].substring( 3) : splitWord[1].substring( 3);
```
Opening parenthesis `(` should not be followed by white space.

nknguyenhc commented on src/main/java/seedu/Task.java:
```diff
@@ -0,0 +1,121 @@
+                    String date = splitWord[1].substring( 3).length() == 1 ? "0"
+                            + splitWord[1].substring( 3) : splitWord[1].substring( 3);
+
+                    String endDate = splitWord[3]+"-"+splitWord[2]+"-" + date;
```
There should be space before and after `+` operator.

nknguyenhc commented on src/main/java/seedu/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+                Task t = tasks.get(i);
+                String start = !t.start.isEmpty() ? t.start + "|" : "";
+                String end = !t.end.isEmpty()?  t.end + "|" : "";
+                arrays.add(t.category.toString() + "|" + (t.isDone ? "true" : "false") + "|" + t.title + "|" + end + start);
```
Line is too long, you may want to consider breaking the line (with appropriate indentation).

nknguyenhc commented on build.gradle:
```diff
@@ -0,0 +1,68 @@
+    modules = [ 'javafx.controls', 'javafx.fxml' ]
+}
+application {
+    mainClass.set("seedu.Duke")
```
Should indicate the correct package, and the static `main` method in the class should be runnable.

nknguyenhc reviewed (commented):
Generally, there are quite a few coding standard issues. You can run `gradlew checkstyleMain` to detect these issues and fix accordingly.

Packaging can be improved. `seedu` is the default package name given from the template, but you do not need to follow it. You may want to organise your Java classes into the correct folder and package them accordingly.

Vanessamae23 committed:
setDuke method does not have any javadoc initially

Added the javadoc header details to give more comprehensive details
about what the method is doing

Vanessamae23 committed:
Added some assertions

Vanessamae23 committed:
Refactoring done

Vanessamae23 committed:
Merge pull request #1 from Vanessamae23/Branch-Assertion

Added some assertions

Vanessamae23 committed:
Merge branch 'Branch-CodeQuality'

Vanessamae23 committed:
Add extension

Vanessamae23 committed:
Added extension:

Vanessamae23 committed:
Add better gui

Vanessamae23 committed:
Added better gui

Vanessamae23 committed:
Added UI image

Vanessamae23 committed:
Add UI image

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Update README.md

Vanessamae23 committed:
Amend the IP

Vanessamae23 committed:
Merge branch 'master' of https://github.com/Vanessamae23/cs2103t-ip
